### PR TITLE
Extend session expires/expired messages.

### LINKED ISSUE #6740: Banner text: "Your session will expire soon" is not informative/helpful
I got the banner by leaving an open admin tab for a while, and was unsure:

- what session the message was talking about (i did not explicitly start a session) - what is soon?
- what should I do when I see this warning? - not very actionable, is there any way to reset session without losing data?
- what are the consequences of an expired session?

I believe the Code is here:
https://github.com/dart-lang/pub-dev/blob/7707373028394fae78e046a6196cb4e745919eb6/pkg/web_app/lib/src/account.dart#L58

image

### PR DESCRIPTION
**Pull request by isoos**
Fixes #6740.

### CODE REVIEW

**Review by sigurdm — APPROVED** [review]

**Comment by sigurdm** [comment]
I would still be a bit surprised as a user to know that I even initiated a session

Is it true that the session starts when going to the admin page?

File: pkg/web_app/lib/src/account.dart [hunk 1]
```diff
@@ -52,10 +52,12 @@ void _initSessionMonitor() {
     final authenticatedAt = status.authenticatedAt;
     String? displayMessage;
     if (expires == null || authenticatedAt == null) {
-      displayMessage = 'Your session has expired.';
+      displayMessage =
+          'Your session has expired: you may need to reload the page to resume editing.';
```
Comment (sigurdm) [inline pkg/web_app/lib/src/account.dart:56]: Could it be?
```suggestion
          'Your admin session has expired: you may need to reload the page to resume editing.';
```

Comment (sigurdm) [reply pkg/web_app/lib/src/account.dart:56]: Don't you **have** to reload?
```suggestion
          'Your session has expired: reload the page before making further administrative actions.';
```

**Comment by isoos** [comment]
> Don't you have to reload?

I don't think we've tested it yet, but the session handling is designed in a way that you could have opened two tabs, and if you are active on one tab, the other won't expire... so if your tab is closing on expiry, but you resume activity on the other, the next session status update refresh API call in the background would reflect that, and the message will disappear.

**Comment by sigurdm** [comment]
> I don't think we've tested it yet, but the session handling is designed in a way that you could have opened two tabs, and if you are active on one tab, the other won't expire... so if your tab is closing on expiry, but you resume activity on the other, the next session status update refresh API call in the background would reflect that, and the message will disappear.

Let's try having a more clear messaging, and not according for an extremely rare case of two admin tabs, one keeping the other alive.

Just reloading the tab with the warning will always restart the session? I think that serves the 99.9%

Btw. Would it be possible to insert a link in the banner that reloads the session without a page-reload?

**Comment by sigurdm** [comment]
Yeah - we probably should not spend too much time here.

Thanks for improving the message!

**Pull request merged**